R/ensure-source.R

```diff
@@ -180,17 +148,98 @@ ensure_dataset <- function(name, download = TRUE) {
   ds
 }
 
-known_datasets <- list(
-  taxi_parquet = list(
-    url = "s3://ursa-labs-taxi-data",
-    download = function(path) {
-      arrow::copy_files("s3://ursa-labs-taxi-data", path)
-      invisible(path)
-    },
-    open = function(path) {
-      arrow::open_dataset(path, partitioning = c("year", "month"))
-    },
-    dim = c(1547741381L, 20L),
-    n_files = 125
-  )
-)
+source_filename <- function(name) {
+  filename <- get_source_attr(name, "url")
+
+  # if the filename is NULL, this is a test data source
+  if (is.null(filename)) {
+    filename <- get_source_attr(name, "filename")
+  }
+
+  ext <- file_ext(basename(filename))
+  paste(c(name, ext), collapse = ".")
+}
+
+#' Ensure that a source has a specific format
+#'
+#' @param name name of the known source
+#' @param format format to be ensured
+#' @param compression compression to be ensured
+#'
+#' @return the file that was ensured to exist
+#' @export
+#'
+#' @importFrom utils write.csv
+ensure_format <- function(
+  name,
+  format = c("csv", "parquet", "feather", "csv.gz"),
+  compression = c("uncompressed", "snappy", "zstd", "gzip", "gz", "lz4")) {
```

Comment:
I don't think you want both "gzip" and "gz" here. Too many places in the function check `if (compression == "gzip")` so would fail if you provide "gz". 

For better or worse, `match.arg()` will just partial-match `gz` to `gzip` anyway, so even if there were a good reason to support passing `gz` as an argument, it will still work.

```suggestion
  compression = c("uncompressed", "snappy", "zstd", "gzip", "lz4")) {
```

Reply:
Yeah, I only had it because I have the special case `.csv.gz` later and that was on my mind.